projects/packages/forms/src/blocks/shared/hooks/use-form-style-outline-classes-and-styles.js

```diff
@@ -0,0 +1,137 @@
+import { isNumber, merge } from 'lodash';
+import { useFormStyle, FORM_STYLE } from '../../contact-form/util/form';
+
+/**
+ * Returns the value of the CSS var if it is a number, otherwise null.
+ * Match behaviour in projects/packages/forms/src/blocks/shared/hooks/use-jetpack-field-styles.js.
+ *
+ * @param {*} value - A value from the legacy form block attributes.
+ * @return {string|*} The value of the CSS var if it is a number, otherwise the value itself.
+ */
+function getIntAsPxValue( value ) {
+	if ( typeof value !== 'undefined' && isNumber( value ) ) {
+		return `${ value }px`;
+	}
+	return value;
+}
+
+export default function useFormStyleOutlineClassesAndStyles( {
+	clientId,
+	inputBlockName,
+	inputBlockAttributes,
+} ) {
+	const formStyle = useFormStyle( clientId );
+	const { userConfig, baseConfig } = useSelect( select => {
+		const {
+			__experimentalGetCurrentGlobalStylesId,
+			getEditedEntityRecord,
+			__experimentalGetCurrentThemeBaseGlobalStyles,
+		} = select( coreStore );
+		const globalStylesId = __experimentalGetCurrentGlobalStylesId();
+		if ( ! globalStylesId ) {
+			return null;
+		}
+
+		return {
+			userConfig: getEditedEntityRecord( 'root', 'globalStyles', globalStylesId ),
+			baseConfig: __experimentalGetCurrentThemeBaseGlobalStyles(),
+		};
+	}, [] );
+
+	const mergedGlobalBlockStyles = useMemo(
+		() =>
+			merge(
+				baseConfig?.styles?.blocks?.[ inputBlockName ],
+				userConfig?.styles?.blocks?.[ inputBlockName ]
+			),
+		[ baseConfig?.styles?.blocks, userConfig?.styles?.blocks, inputBlockName ]
+	);
+
+	return useMemo( () => {
+		// Only return styles for outlined and animated forms.
+		if ( formStyle !== FORM_STYLE.OUTLINED && formStyle !== FORM_STYLE.ANIMATED ) {
+			return null;
+		}
+		// Access the input block's attributes.
+		const blockBorderClassesAndStyles = getBorderClassesAndStyles( inputBlockAttributes ?? {} );
+		const globalBorderClassesAndStyles = getBorderClassesAndStyles( {
+			style: mergedGlobalBlockStyles?.[ inputBlockName ],
+		} );
+
+		// Notched HTML only needs the background color and associated classes.
+		const attributesWithBackgroundColor = inputBlockAttributes
+			? {
+					backgroundColor: inputBlockAttributes?.backgroundColor,
+					style: {
+						color: {
+							background: inputBlockAttributes?.style?.color?.background,
+						},
+					},
+			  }
+			: {};
+		const blockColorClassesAndStyles = getColorClassesAndStyles( attributesWithBackgroundColor );
+
+		/**
+		 * Remove undefined classname values.
+		 */
+		const filteredBlockColorClassesAndStyles = [
+			blockBorderClassesAndStyles?.className,
+			blockColorClassesAndStyles?.className,
+		]
+			.filter( Boolean )
+			.join( ' ' );
+
+		let styleSpecificCssVars = {};
+
+		if ( formStyle === FORM_STYLE.OUTLINED ) {
+			styleSpecificCssVars = {
+				'--jetpack--contact-form--notch-width':
+					'max(var(--jetpack--contact-form--input-padding-left, 16px), var(--jetpack--contact-form--border-radius))',
```

Comment:
`--jetpack--contact-form--border-radius` needs to be set above to offset any increase in border radius for the notch.

<img width="719" alt="Screenshot 2025-05-09 at 11 06 48 am" src="https://github.com/user-attachments/assets/33f313e6-0fff-4084-9e0b-05021b26caa3" />

Or we could just use the raw value calculated below. I'll have a look.

Reply:
Actually, scrap that sorry. I think it's because of the switch from deepmerge not properly merging global styles, or something related. I'll look 👀

Reply:
>  I think it's because of the switch from deepmerge not properly merging global styles

Sorry if my suggestion led you astray! Feel free to add the dependency back in if it fixes things up!

Reply:
No worries!! Probably an oversight on my part I'd wager.

Still convinced we could use `_.merge`...

Reply:
Yep, totally my fault 🤣 Fixing.